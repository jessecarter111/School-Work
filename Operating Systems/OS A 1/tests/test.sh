#!/bin/sh

PREFIX=test
TESTS='0 1 2 3 4 5 6 7 8 9 10 11 12 13 14'
PROG=runme.sh
count=0

if [ ! -x $PROG ]; then
  echo Error: $PROG is not in the current directory or is not executable
  exit 1
fi

for T in $TESTS; do
  if [ -f $PREFIX.$T.xo ]; then  
    echo ===========================================
    echo Test file: $PREFIX.$T.xo 
    ./$PROG 100000 $PREFIX.$T.xo > $PREFIX.$T.out
    if diff $PREFIX.$T.out $PREFIX.$T.gold > /dev/null; then
      echo " " PASSED
      let "count = count + 1"
    else
      echo " " FAILED
    fi
    # rm $PREFIX.$T.out
  else
    echo Oops: file $PREFIX.$T.xo does not exist!
  fi
done

echo =============================================
echo $count tests passed
