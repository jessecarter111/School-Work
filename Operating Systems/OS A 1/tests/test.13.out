00> PC: 0002, State: 0002
00> Registers: 
00>      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 72, r0

00> PC: 0006, State: 0002
00> Registers: 
00>      r00:0048 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0

H00> PC: 0008, State: 0002
00> Registers: 
00>      r00:0048 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 101, r0

00> PC: 000c, State: 0002
00> Registers: 
00>      r00:0065 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0

e00> PC: 000e, State: 0002
00> Registers: 
00>      r00:0065 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 108, r0

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:006c r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0

l00> PC: 0014, State: 0002
00> Registers: 
00>      r00:006c r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 108, r0

00> PC: 0018, State: 0002
00> Registers: 
00>      r00:006c r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0

l00> PC: 001a, State: 0002
00> Registers: 
00>      r00:006c r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 111, r0

00> PC: 001e, State: 0002
00> Registers: 
00>      r00:006f r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0

o00> PC: 0020, State: 0002
00> Registers: 
00>      r00:006f r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 10, r0

00> PC: 0024, State: 0002
00> Registers: 
00>      r00:000a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: out r0


00> PC: 0026, State: 0002
00> Registers: 
00>      r00:000a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: ret 

00> PC: 0300, State: 0002
00> Registers: 
00>      r00:000a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0002
00> Next instruction: (null) 

